### PR TITLE
feat(e2e): custom-firmware

### PR DESCRIPTION
**Pull request by STew790**
custom fw upload test

### CODE REVIEW

File: packages/integration-tests/projects/suite-web/tests/settings/custom-firmware.test.ts [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+        cy.task('setupEmu');
+        cy.task('startBridge');
+        cy.viewport(1080, 1440).resetDb();
+        cy.prefixedVisit('/settings/device');
+        cy.passThroughInitialRun();
+    });
+    /*
+    *1. Navigate to `Settings/Device`
+    *2. Click on `Install firmware`
+        a)Install custom firmware modal appears
```
Comment (ondracja) [inline packages/integration-tests/projects/suite-web/tests/settings/custom-firmware.test.ts]: I would skip the `expected result` part of the test case, thats for cypresse's assert to verify.

**Review by ondracja — COMMENTED** [review]
I would also rename the test bin file, this one looks sus. How about `testFirmware.bin` ?

**Comment by ondracja** [comment]
And last thing - pls rebase the branch so there's only one commit.

**Review by ondracja — APPROVED** [review]

File: packages/integration-tests/projects/suite-web/tests/settings/custom-firmware.test.ts [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+        //
+        // Test preparation
+        //
+        cy.getTestElement('@settings/device/custom-firmware-modal-button').click({
+            force: true,
+        });
+        //
+        // Test execution
+        //
+        cy.get('[data-test="@firmware-custom"]').then(fileUploadModal => {
```
Comment (mroz22) [inline packages/integration-tests/projects/suite-web/tests/settings/custom-firmware.test.ts]: using 
```
cy.getTestElement("@firmware-custom").then(...)
```
does not work here? maybe it indeed does not. If so, we should fix it. cy.getTestElement should have the same function signature as cy.get

File: packages/integration-tests/projects/suite-web/tests/settings/custom-firmware.test.ts [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+                .find('input[type=file]')
+                .attachFile('test.bin', { subjectType: 'drag-n-drop' });
+        });
+        cy.get('[data-test="@firmware-custom"]').then(installFWbutton => {
+            cy.wrap(installFWbutton)
+                .find('[class*="Button"]')
+                .should('not.be.disabled')
+                .contains('Install firmware')
+                .click();
+        });
```
Comment (mroz22) [inline packages/integration-tests/projects/suite-web/tests/settings/custom-firmware.test.ts:41]: idea for improvement: would be great to add some screenshots here. You can use 
`cy.screenshot(name)` which does not do any assertions, only produces screenshots or `cy.matchImageSnapshots(name)`

Comment (ondracja) [reply packages/integration-tests/projects/suite-web/tests/settings/custom-firmware.test.ts:41]: @STew790 , after a discussion with @mroz22, we agreed that for sanity tests, it does not make sense to make and maintain screenshots as there is usually only one assert. So when dealing with sanity tests, let's keep using chai assertions unless there is a very good individual use case.

**Comment by ondracja** [comment]
Closing this pull request because of a mistake I made with yarn.lock. Will transfer the code to another branch and PR.

**Pull request closed without merging**